### ISSUE TITLE
Current write implementation in UART HAL may not write 100% of data

### ISSUE THREAD

**Issue by greer184** [kubos/kubos#76]
In the write implementation of the Stream in the Rust UART HAL I wrote in #68, a Result with an empty tuple is returned. However, the underlying port write function returns a result with a ``usize`` variable. The issue here is that a result may return without an error even though some of the bytes may not have been written to across the port. 

We recommend simply adjusting the function call to match the expected behavior of our write function. In the serial_unix API, there is a ``write_all()`` function that performs an all or nothing write which matches what we are trying do in our write function. This change shouldn't affect anything currently calling the function.

Basically, we change this line:
``port.write(data)?;``

To this line:
``port.write_all(data)?;``
